OpenRA.Game/FileFormats/AudLoader.cs

```diff
@@ -153,7 +166,33 @@ public static byte[] LoadSound(Stream s)
 				dataSize -= 8 + chunk.CompressedSize;
 			}
 
-			return output;
+			rawData = output;
+			return true;
+		}
+
+		public bool TryParseSound(Stream stream, string fileName, out byte[] rawData, out int channels, out int sampleBits,
+			out int sampleRate)
+		{
+			channels = sampleBits = sampleRate = 0;
+
+			try
+			{
+				if (!LoadSound(stream, out rawData))
+					return false;
+			}
+			catch (Exception e)
+			{
+				Log.Write("debug", "Failed to parse AUD file {0}. Error message:".F(fileName));
```

Comment:
Actually no.
The inner parsing method (`LoadSound()` in this case) returns `true`/`false` depending on whether the file is in a format this parser accepts. So the `AudLoader` will just return `false` for WAV files and vice-versa instead of throwing.
The exceptions happen when the file is corrupt or otherwise bogus - ends prematurely, the metadata at the start doesn't correspond to the actual data (like in #10124) or other such situations. So only actual exceptions will be logged.

Perhaps I should add a comment to make that clear by just glancing at the code.